packages/babel-plugin-transform-object-rest-spread/src/index.js

```diff
@@ -21,23 +21,69 @@ export default function({ types: t }) {
     return false;
   }
 
-  function createObjectSpread(file, props, objRef) {
-    const restElement = props.pop();
-
+  // normalizes identifier keys to string literals and extracts all keys
+  // e.g. extracts ["a", "b", 3, ++x] from {a: "foo", b, 3: "bar", [++x]: "baz"}
+  function extractNormalizedKeys(path) {
+    const propPaths = path.get("properties");
     const keys = [];
-    for (const prop of props) {
-      let key = prop.key;
-      if (t.isIdentifier(key) && !prop.computed) {
-        key = t.stringLiteral(prop.key.name);
+    for (const propPath of propPaths) {
+      const key = propPath.get("key").node;
+      if (t.isIdentifier(key) && !propPath.node.computed) {
+        keys.push(t.stringLiteral(key.name));
+      } else {
```

Comment:
Adding to the previous comment, we'd also have to worry about computed properties which evaluate to a number or any other literal than the string literal. We'd still need to call `toPropertyKey` for those too.

But I guess converting to `stringLiterals`, we could make the transform output simpler for some cases. So that can be added.